### PR TITLE
test python version of projectAreaRename script

### PR DESCRIPTION
**Pull request by tulamor**

### CODE REVIEW

**Comment by smuzaffar** [comment]
@tulamor , this is not working. It is deleting all files under config directory.

```
scram p CMSSW_11_2_X_2020-07-03-1200
cd CMSSW_11_2_X_2020-07-03-1200/
cmsenv
git cms-addpkg FWCore/Common
wget -o config/SCRAM/projectAreaRename.pl https://raw.githubusercontent.com/cms-sw/cmssw-config/1484a40fdf49685579b859d50c13ed803b1ccb21/SCRAM/projectAreaRename.py
scram b echo_CXX
cd ..
mkdir FOO
mv CMSSW_11_2_X_2020-07-03-1200 FOO/
cd FOO/CMSSW_11_2_X_2020-07-03-1200
scram b ProjectRename
ls config -l
-rw-r--r--. 1 muzaffar zh    0 Jul  3 18:30 BuildFile.xml
-rw-r--r--. 1 muzaffar zh    0 Jul  3 18:30 ChangeLog
-rw-r--r--. 1 muzaffar zh    0 Jul  3 18:30 README.md
drwxr-xr-x. 5 muzaffar zh 4096 Jul  3 18:30 SCRAM
-rw-r--r--. 1 muzaffar zh    0 Jul  3 18:30 SCRAM_ExtraBuildRule.pm
-rw-r--r--. 1 muzaffar zh    0 Jul  3 18:30 Self.xml
-rw-r--r--. 1 muzaffar zh    0 Jul  3 18:30 boot.xml
-rw-r--r--. 1 muzaffar zh    0 Jul  3 18:30 bootsrc.xml
-rw-r--r--. 1 muzaffar zh    0 Jul  3 18:30 config_tag
-rw-r--r--. 1 muzaffar zh    0 Jul  3 18:30 dxr.config
-rwxr-xr-x. 1 muzaffar zh    0 Jul  3 18:30 fixdocs.sh
-rwxr-xr-x. 1 muzaffar zh    0 Jul  3 18:30 genClassesH
-rw-r--r--. 1 muzaffar zh    0 Jul  3 18:30 scram_basedir
-rw-r--r--. 1 muzaffar zh    0 Jul  3 18:30 scram_version
drwxr-xr-x. 3 muzaffar zh 4096 Jul  3 17:07 toolbox
-rw-r--r--. 1 muzaffar zh    0 Jul  3 18:30 updateConfig.py
```

File: SCRAM/projectAreaRename.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+                gzip = 1
+                s = stat(f)
+                ftime = s[9]
+                call("gunzip -S.gz %s" % f, shell=True)
+                f = re.sub(r'.gz$', r'', f)
+            has_olddir = 0
+            try:
+                with open("%s.rename" % f, "w+") as oref:
+                    try:
+                        with open(f, "w+") as iref:
```
Comment (smuzaffar) [inline SCRAM/projectAreaRename.py:37]: this should be open in read only mode 
```
with open(f) as iref:
```
is enough

File: SCRAM/projectAreaRename.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            gzip = 0
+            if re.search('.db.gz$', f):
+                gzip = 1
+                s = stat(f)
+                ftime = s[9]
+                call("gunzip -S.gz %s" % f, shell=True)
+                f = re.sub(r'.gz$', r'', f)
+            has_olddir = 0
+            try:
+                with open("%s.rename" % f, "w+") as oref:
```
Comment (smuzaffar) [inline SCRAM/projectAreaRename.py:35]: why opening the file in read/write mode. Just "w" should be enough

File: SCRAM/projectAreaRename.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python
```
Comment (smuzaffar) [inline SCRAM/projectAreaRename.py:1]: ifI change this to python3 then it fails. please check this for python3 as well

File: SCRAM/projectAreaRename.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+rel = args.dir
+while not path.isdir(path.join(rel, ".SCRAM")) and re.search(r'^[\.\/]$', rel):
+    rel = path.dirname(rel)
+if not path.isdir(path.join(rel, ".SCRAM")):
+    raise Exception("%s is not a SCRAM-based project area." % args.dir)
+
+if args.olddir != args.newtop:
+    qod = re.escape(args.olddir)
+    for d in [path.join(rel, ".SCRAM", args.arch), path.join(rel, "config")]:
+        for f in check_output("find %s -type f" % d, shell=True).rstrip().splitlines():
```
Comment (smuzaffar) [inline SCRAM/projectAreaRename.py:24]: should be `.decode().rstrip().splitlines()`

File: SCRAM/projectAreaRename.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+            else:
+                if ftime == 0:
+                    s = stat(f)
+                    ftime = s[9]
+                shutil.move("%s.rename" % f, f)
+            if gzip:
+                call("gzip -S.gz %s" % f, shell=True)
+                f = "%s.gz" % f
+            if ftime > 0:
+                utime(f,(ftime, ftime))
```
Comment (smuzaffar) [inline SCRAM/projectAreaRename.py:58]: setting of time is not working. at the end files shoudl have the original timestamp

**Comment by tulamor** [comment]
@smuzaffar, there is one file which differs: .SCRAM/slc7_amd64_gcc820/DirCache.db

```
diff CMSSW_11_2_X_2020-07-01-2300.PERL/.SCRAM/slc7_amd64_gcc820/DirCache.db CMSSW_11_2_X_2020-07-01-2300.PY/.SCRAM/slc7_amd64_gcc820/DirCache.db
6c6
<                                                                        1594045858
---
>                                                                        1594046155
9c9
<                                                                             1594045857
---
>                                                                             1594046155
12c12
<                                                                          1594045858,
---
>                                                                          1594046156,
19c19
<                                                                        1594045858
---
>                                                                        1594046156
22c22
<                                                                        1594045858
---
>                                                                        1594046156
25c25
<                                                                        1594045857,
---
>                                                                        1594046155,
```


is it acceptable or it is a bug?

**Comment by smuzaffar** [comment]
yes this is fine

**Comment by smuzaffar** [comment]
this looks good, I will integrated it for DEVEL IBs to fully test it.

**Pull request merged**